### PR TITLE
fix: restful prepare statement can support preprocessSql

### PR DESCRIPTION
**Pull request by clcl2**
connect can support preprocessSql

### CODE REVIEW

**Review by clcl2 — COMMENTED** [review]
restfulprepareStatement  can support preprocessSql

**Review by huolibo — APPROVED** [review]
LGTM

File: src/main/java/com/taosdata/jdbc/utils/Utils.java [hunk 1]
```diff
@@ -230,4 +234,31 @@ private static boolean isNanoSecFormat(String timestampStr) {
...
+    /**
+     * Some of the SQLs sent by other popular frameworks or tools like Spark, contains syntax that cannot be parsed by
+     * the TDengine client. Thus, some simple parsers/filters are intentionally added in this JDBC implementation in
+     * order to process those supported SQLs.
+     */
+    public static String preprocessSql(String rawSql) {
+        //For processing some of Spark SQLs
+        // SELECT * FROM db.tb WHERE 1=0
+        rawSql = rawSql.replaceAll("WHERE 1=0", "WHERE _c0=1");
+        rawSql = rawSql.replaceAll("WHERE 1=2", "WHERE _c0=1");
```
Comment (sangshuduo) [inline src/main/java/com/taosdata/jdbc/utils/Utils.java:247]: _c0 is the first timestamp column. 1 is a valid value if the users intend to store very old data (around the Unix epoch). So the statement has a small chance to fail.

**Review by sangshuduo — CHANGES_REQUESTED** [review]
see my comments

File: src/main/java/com/taosdata/jdbc/utils/Utils.java (hunk 1 above)
Comment (sangshuduo) [reply src/main/java/com/taosdata/jdbc/utils/Utils.java:247]: suggest to use "WHERE _c0 is null" to replace

**Review by sangshuduo — CHANGES_REQUESTED** [review]
see my comment

**Review by sangshuduo — APPROVED** [review]
LGTM

File: src/main/java/com/taosdata/jdbc/utils/Utils.java (hunk 1 above)
Comment (clcl2) [reply src/main/java/com/taosdata/jdbc/utils/Utils.java]: thanks

**Review by dingbo8128 — APPROVED** [review]
LGTM

**Pull request merged**